fs/inode/fs_files.c

```diff
@@ -51,59 +51,118 @@
  * Private Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: files_fget_index
+ ****************************************************************************/
+
+static FAR struct file *files_fget_index(FAR struct filelist *list,
+                                         int l1, int l2)
+{
+  FAR struct file *filep;
+  irqstate_t flags;
+
+  flags = enter_critical_section();
```

Comment:
shouldn't we use spin_lock_irqsave here and below ?

Reply:
`critical_section` has been protected by spin lock in the case of SMP

Reply:
I think @raiden00pl suggest that spinlock can be faster in SMP case, so it's better to change mutex to spinlock_t

Reply:
Done